style-guide/README.md

```diff
@@ -0,0 +1,187 @@
+# Rust Style Guide
+
+## Motivation - why use a formatting tool?
+
+Formatting code is a mostly mechanical task which takes both time and mental
+effort. By using an automatic formatting tool, a programmer is relieved of
+this task and can concentrate on more important things.
+
+Furthermore, by sticking to an established style guide (such as this one),
+programmers don't need to formulate ad hoc style rules, nor do they need to
+debate with other programmers what style rules should be used, saving time,
+communication overhead, and mental energy.
+
+Humans comprehend information through pattern matching. By ensuring that all
+Rust code has similar formatting, less mental effort is required to comprehend a
+new project, lowering the bar to entry for new developers.
+
+Thus, there are productivity benefits to using a formatting tool (such as
+rustfmt), and even larger benefits by using a community-consistent formatting,
+typically by using a formatting tool's default settings.
+
+
+## Formatting conventions
+
+### Indentation and line width
+
+Use spaces, not tabs. Each level of indentation must be four spaces. The maximum
```

Comment:
> Each level of indentation must be four spaces.

If this means what I think it means, it would be good to get a totally incontrovertible statement of it somewhere around here. Something like: **other than inside of a multiline string literal, the number of leading spaces on a line must always be a multiple of 4.**

As currently worded, I can imagine disagreement about whether:

```rust
(abcd,
 defg)
```

constitutes "a level of indentation."